Use account.charges_enabled to show if Stripe payments are available

EvanDotPro opened the pull request:
Partially fixes #2175. This fixes the "Payments" status but #2175 also has people mentioning that the "Payouts" status is showing "Disabled" incorrectly, which this PR does not address.

## Changes proposed in this Pull Request:

According to Stripe, due to some historical inconsistencies in their systems, the `card_payments` capability is not always returned for accounts of certain ages or circumstance, even if those accounts are in fact able to receive card payments.

This pull request changes how the "Payments: Enabled/Disabled" status is shown in the "Account details" panel within the Stripe settings. Instead of checking for the `card_payments` capability, we now simply check for the `charges_enabled` Stripe account property.

Additionally, some merchants have payouts "enabled" but do not have a payout schedule configured in Stripe. However prior to this fix, the Stripe settings would only show payouts as enabled if the merchant had a payout schedule configured in Stripe, even if manual payouts were enabled/available.

This fix also makes the Stripe settings show payouts as Enabled if the `payouts_enabled` property is true in their Stripe account even if they do not have a payout schedule set up.

## Screenshots

Before fix when Stripe account has `charges_enabled`: `true` but does not have the `card_payments` capability:

![](https://caps.evan.pro/1e44d2.png)

After fix when Stripe account has `charges_enabled`: `true` but does not have the `card_payments` capability:

![](https://caps.evan.pro/e8b26c.png)

## Testing instructions

This is particularly tricky to test because to _truly_ test it requires a Stripe account created before a certain date or under some specific unknown circumstances.

However, a hacky way to fake the condition is to temporarily add `unset( $account->capabilities );` [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/137b14f6896e369d0d8265b34d66ec6474772e28/includes/class-wc-stripe-account.php#L88) then clear the `_transient_wcstripe_account_data_test` and `_transient_wcstripe_account_data_live` transients from the db.

With that hack in place, before this patch it will show Payments: Disabled... After the patch it will show Payments: Enabled.

EvanDotPro committed:
Use account.charges_enabled to show if Stripe payments are available.

Partially fixes #2175

EvanDotPro committed:
Added 'charges_enabled' Stripe account property to JS mocks

This is a continuation to fixes for #2175. Also removed now unneeded
mock(s).

ricardo reviewed (approved):
LGTM. Tests well. 👏

Edit: Do we need a changelog entry before merging?

EvanDotPro committed:
Don't check for payout schedule to determine if Stripe payouts enabled

Fixes #2175. Some merchants have payouts "enabled" but do not have a
payout schedule configured in Stripe -- presumably they are triggering
their payouts manually in these cases.

This change makes the Stripe settings show payouts as Enabled if the
`payouts_enabled` property is true in their Stripe account even if they
do not have a payout schedule set up.

EvanDotPro committed:
Add changelog entry for #2175 fix

EvanDotPro commented:
Going to rebase this after #2348 is merged, which should get all the checks green for this PR.
